File: app/src/main/java/com/test/your_movie/NetworkService.java
package com.test.your_movie;

import android.support.annotation.NonNull;
import android.util.Log;

import com.test.your_movie.Interfaces.JSONPlaceHolderApi;
import com.test.your_movie.Models.Movie;
import com.test.your_movie.Models.JSONMovieResponse;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Date;
import java.util.List;
import java.util.Locale;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class NetworkService {
    private static NetworkService mInstance;
    private static final String BASE_URL = "https://api.themoviedb.org";
    private Retrofit mRetrofit;
    private LiveDataHelper liveDataHelper;
    private List<Movie> movieList = new ArrayList<>();


    private NetworkService() {
        mRetrofit = new Retrofit.Builder().baseUrl(BASE_URL).addConverterFactory(GsonConverterFactory.create()).build();
        liveDataHelper = LiveDataHelper.getInstance();
    }

    public static NetworkService getInstance() {
        if (mInstance == null) {
            mInstance = new NetworkService();
        }
        return mInstance;
    }

    public JSONPlaceHolderApi getJSONApi() {
        return mRetrofit.create(JSONPlaceHolderApi.class);
    }

    public void loadJsonData(int pageNumber){
        final String TAG = "LoadJSONData";

        NetworkService.getInstance()
                .getJSONApi()
                .listRepos(pageNumber)
                .enqueue(new Callback<JSONMovieResponse>() {
                    @Override
                    public void onResponse(@NonNull Call<JSONMovieResponse> call, @NonNull Response<JSONMovieResponse> response) {
                        JSONMovieResponse jsonMovieResponse = response.body();

                        if(jsonMovieResponse != null) {
                            List<Movie> newList = new ArrayList<>(Arrays.asList(jsonMovieResponse.getMovieArray()));

                            for (Movie m : newList) {
                            SimpleDateFormat from = new SimpleDateFormat("yyyy-mm-dd", Locale.UK);
                            SimpleDateFormat to = new SimpleDateFormat("d MMMM yyyy", Locale.UK);
                            Date date = null;
                            try {
                                date = from.parse(m.getReleaseDate());
                            } catch (ParseException e) {
                                e.printStackTrace();
                            }
                            m.setReleaseDate(to.format(date));
                            }

                            movieList.addAll(newList);
                            liveDataHelper.updateMovieList(movieList);

                        } else {
                            Log.i(TAG, "null objects " + response.toString());
                        }
                    }

                    /**
                     * Unsuccessful connection calls the NoInternetFragment
                     */
                    @Override
                    public void onFailure(@NonNull Call<JSONMovieResponse> call, @NonNull Throwable t) {
                        Log.i("JsonCall", "Error occurred while getting request!");
//                        loadNoInternetFragment();
                        t.printStackTrace();
                    }
                });
    }
}

File: app/src/main/java/com/test/your_movie/Interfaces/NetworkService.kt
package com.test.your_movie.Interfaces

import com.test.your_movie.network.model.movie.Movie

import java.util.ArrayList

class NetworkService private constructor() {
//    private val mRetrofit: Retrofit
//    private val liveDataHelper: LiveDataHelper
    private val movieList = ArrayList<Movie>()

//    val jsonApi: JSONPlaceHolderApi
//        get() = mRetrofit.create(JSONPlaceHolderApi::class.java)

    init {
//        mRetrofit = Retrofit.Builder().baseUrl(MOVIE_LIST_URL).addConverterFactory(GsonConverterFactory.create()).build()
//        liveDataHelper = LiveDataHelper.Companion.getInstance()
    }

    fun loadJsonData(pageNumber: Int) {
        val TAG = "LoadJSONData"

//        NetworkService.instance
//                .jsonApi
//                .movieList(pageNumber)
//                .enqueue(object : Callback<MovieResponse> {
//                    override fun onResponse(call: Call<MovieResponse>, response: Response<MovieResponse>) {
//                        val jsonMovieResponse = response.body()
//
//                        if (jsonMovieResponse != null) {
//                            val newList = ArrayList(Arrays.asList(*jsonMovieResponse.movieArray!!))
//
//                            for (m in newList) {
//                                val from = SimpleDateFormat("yyyy-mm-dd", Locale.UK)
//                                val to = SimpleDateFormat("d MMMM yyyy", Locale.UK)
//                                var date: Date? = null
//                                try {
//                                    date = from.parse(m.releaseDate!!)
//                                } catch (e: ParseException) {
//                                    e.printStackTrace()
//                                }
//
//                                m.releaseDate = to.format(date!!)
//                            }
//
//                            movieList.addAll(newList)
////                            liveDataHelper.updateMovieList(movieList)
//
//                        } else {
//                            Log.i(TAG, "null objects $response")
//                        }
//                    }
//
//                    /**
//                     * Unsuccessful connection calls the NoInternetFragment
//                     */
//                    override fun onFailure(call: Call<MovieResponse>, t: Throwable) {
//                        Log.i("JsonCall", "Error occurred while getting request!")
//                        //                        loadNoInternetFragment();
//                        t.printStackTrace()
//                    }
//                })
    }

//    companion object {
        private var mInstance: NetworkService? = null
        private val BASE_URL = "https://api.themoviedb.org"
//
//        val instance: NetworkService
//            get() {
//                if (mInstance == null) {
//                    mInstance = NetworkService()
//                }
//                return mInstance
//            }
//    }
}
